hibernate-core/src/main/java/org/hibernate/hql/internal/ast/exec/BasicExecutor.java

```diff
@@ -55,42 +40,41 @@ public BasicExecutor(HqlSqlWalker walker, Queryable persister) {
 	}
 
 	@Override
-	public int execute(QueryParameters parameters, SharedSessionContractImplementor session) throws HibernateException {
-		return doExecute(
-			parameters,
-			session,
-			session.getJdbcServices().getDialect()
-					.addSqlHintOrComment(
-						sql,
-						parameters,
-						session.getFactory().getSessionFactoryOptions().isCommentsEnabled()
-					),
-			parameterSpecifications
-		);
-	}
-	
-	protected int doExecute(QueryParameters parameters, SharedSessionContractImplementor session, String sql,
-			List parameterSpecifications) throws HibernateException {
+	public int execute(QueryParameters parameters, SharedSessionContractImplementor session)
+			throws HibernateException {
+
 		BulkOperationCleanupAction action = new BulkOperationCleanupAction( session, persister );
 		if ( session.isEventSource() ) {
-			( (EventSource) session ).getActionQueue().addAction( action );
+			( (EventSource) session).getActionQueue().addAction( action );
 		}
 		else {
 			action.getAfterTransactionCompletionProcess().doAfterTransactionCompletion( true, session );
 		}
 
-		PreparedStatement st = null;
-		RowSelection selection = parameters.getRowSelection();
+		return doExecute(
+				session.getJdbcServices().getDialect()
+						.addSqlHintOrComment(
+								sql,
+								parameters,
+								session.getFactory().getSessionFactoryOptions().isCommentsEnabled()
+						),
+				parameters,
+				parameterSpecifications,
+				session
+		);
+	}
 
+	int doExecute(String sql, QueryParameters parameters, List<ParameterSpecification> parameterSpecifications, SharedSessionContractImplementor session)
```

Comment:
Why? `BasicExecutor.doExecute()` isn't a public extension point of Hibernate, and the new signature is clearly better.

Reply:
Indeed, it's an internal API, but just like HR uses APIs like this, I do too in Blaze-Persistence. Since we are not adding/removing anything but just reordering, it would be cool if we could revert the signature change, but I'll adapt my code if necessary.